tools/add_triage_signature.py

```diff
@@ -2037,6 +2037,52 @@ def _process_ticket(self, url, issue_key):
             self._update_triaging_ticket("\n".join(host_messages))
 
 
+class ErrorOnCleanupInstallDevice(Signature):
+    """
+    This signature monitors errors on cleanupInstallDevice function
+    """
+
+    # This has to be maintained to be the same format as
+    # https://github.com/openshift/assisted-installer/blob/51c021f3245ef1d9e1a50a3e31dc23350cdc5d32/src/installer/installer.go#L98
+    # Remember to maintain backwards compatibility if that format ever changes - create
+    # PATTERN_NEW and try to match both.
+    LOG_PATTERN = re.compile(r'time=".+" level=[a-z]+ msg="(?P<message>failed to prepare install device.*)"')
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args,
+            **kwargs,
+            comment_identifying_string="h1. Non-fatal error(s) occured trying to perform best-effort cleanup on installation disk",
+        )
+
+    def _process_ticket(self, url, issue_key):
+        cluster = get_metadata_json(url)["cluster"]
+        triage_logs_tar = get_triage_logs_tar(triage_url=url, cluster_id=cluster["id"])
+
+        hosts = []
+        for host in cluster["hosts"]:
+            host_id = host["id"]
+
+            try:
+                installer_logs = get_host_log_file(triage_logs_tar, host_id, "installer.logs")
+            except FileNotFoundError:
+                continue
+
+            m = self.LOG_PATTERN.search(installer_logs)
+            if m:
+                hosts.append(
+                    OrderedDict(
+                        host=self._get_hostname(host),
+                        message=m.group("message"),
```

Comment:
```suggestion
            if match := self.LOG_PATTERN.search(installer_logs):
                hosts.append(
                    OrderedDict(
                        host=self._get_hostname(host),
                        message=match.group("message"),
```

Reply:
didnt, know about := operator in python, sweet!

Changed